tests/vm_userdata_test.go

```diff
@@ -75,20 +78,43 @@ var _ = Describe("CloudInit UserData", func() {
 	Describe("A new VM", func() {
 		Context("with cloudInitNoCloud source", func() {
 			It("should have cloud-init data", func(done Done) {
-				magicStr := "printed from cloud-init userdata"
-				userData := fmt.Sprintf("#!/bin/sh\n\necho '%s'\n", magicStr)
+				userData := fmt.Sprintf("#!/bin/sh\n\necho '%s'\n", expectedUserData)
 
-				vm := tests.NewRandomVMWithEphemeralDiskAndUserdata("kubevirt/cirros-registry-disk-demo:devel", userData)
-				obj := LaunchVM(vm)
-				VerifyUserDataVM(vm, obj, magicStr)
+				vm := tests.NewRandomVMWithEphemeralDiskAndUserdata(tests.RegistryDiskCirros, userData)
+				LaunchVM(vm)
+				VerifyUserDataVM(vm, []expect.Batcher{
+					&expect.BExp{R: expectedUserData},
+				}, time.Second*120)
 				close(done)
 			}, 180)
+
+			Context("with injected ssh-key", func() {
+				It("should have ssh-key under authorized keys", func() {
+					userData := fmt.Sprintf(
+						"#cloud-config\npassword: %s\nchpasswd: { expire: False }\nssh_authorized_keys:\n  - %s",
+						fedoraPassword,
+						sshAuthorizedKey,
+					)
+					vm := tests.NewRandomVMWithEphemeralDiskAndUserdataHighMemory(tests.RegistryDiskFedora, userData)
+
+					LaunchVM(vm)
+
+					VerifyUserDataVM(vm, []expect.Batcher{
+						&expect.BExp{R: "login:"},
+						&expect.BSnd{S: "fedora\n"},
+						&expect.BExp{R: "Password:"},
+						&expect.BSnd{S: fedoraPassword + "\n"},
+						&expect.BExp{R: "$"},
+						&expect.BSnd{S: "cat /home/fedora/.ssh/authorized_keys\n"},
+						&expect.BExp{R: "test-ssh-key"},
+					}, time.Second*240)
+				}, 300)
```

Comment:
On the long run we need a different solution than the fodora cloud image. It takes alone two minutes to boot on my machine. I guess Cirros does not properly read our `#cloud-config`?

Reply:
@rmohr I tried to run it on alpine and cirros, but I saw that it does not add any ssh keys(maybe just because that ssh package does not exist or .ssh directory does not exist)
I can take another look on it.

Reply:
@rmohr I checked cirros image and it does not have cloud-init command or cloud-init directories under `/etc`, I do not sure how it runs our `/bin/bash` scripts, maybe it has some very limited cloud-init functional, but it does not suitable for more complex cloud-init checks.
For example under our `cluster/vm-nocloud.yaml` we use user-data with content
```
#cloud-config
password: atomic
ssh_pwauth: True
chpasswd: { expire: False }
```
but I can not login to cirros image with `atomic` password(tried for root and cirros users)

Reply:
@cynepco3hahue  I can confirm that cirros won't work with the standard cloud-init ssh related options in cloud-config.

If you want to get cirros to work with ssh, you'd have to do this in cloud-init.

```
#!/bin/bash
export NEW_USER="foo"
export SSH_PUB_KEY="ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC6zdgFiLr1uAK7PdcchDd+LseA5fEOcxCCt7TLlr7Mx6h8jUg+G+8L9JBNZuDzTZSF0dR7qwzdBBQjorAnZTmY3BhsKcFr8Gt4KMGrS6r3DNmGruP8GORvegdWZuXgASKVpXeI7nCIjRJwAaK1x+eGHwAWO9Z8ohcboHbLyffOoSZDSIuk2kRIc47+ENRjg0T6x2VRsqX27g6j4DfPKQZGk0zvXkZaYtr1e2tZgqTBWqZUloMJK8miQq6MktCKAS4VtPk0k7teQX57OGwD6D7uo4b+Cl8aYAAwhn0hc0C2USfbuVHgq88ESo2/+NwV4SQcl3sxCW21yGIjAGt4Hy7J $NEW_USER@localhost.localdomain"

sudo adduser -U -m $NEW_USER
echo "$NEW_USER:atomic" | chpasswd
sudo mkdir /home/$NEW_USER/.ssh
sudo echo "$SSH_PUB_KEY" > /home/$NEW_USER/.ssh/authorized_keys
sudo chown -R ${NEW_USER}: /home/$NEW_USER/.ssh

```

Reply:
@davidvossel It looks like call to regular bash script, I prefer to stick to more common usage of cloud-init(RHEL, Fedora, CentOS)